### DISCUSSION TITLE
Add average to History (as a line chart on top of the bar charts)

### DISCUSSION THREAD

**Discussion by ghost** [iSoron/uhabits#2038] [Feature Requests]
I have a habit for meditation. I track number of minutes per day. I like that I am successful reaching my minimum number, but would like to know the average number per day total or in a week, or month, or eventually a year. 

My suggestion is to add a line over the bar charts. For example:
- In day view show the average for all days, it could be a straight line that reads 47 minutes
- in week view, show the average daily # of minutes for each week. So the line would go up and down on top of each bar. One week it would be 31 mins, the next 64 mins, etc.
- in the month view, again each month would show the daily average for that month and the line would go up or down. I could see if on average I meditated a lot less in summer months.

Details about how it could work:
- The axis for the line chart would have to be separate from the bar axis max. Like in PowerPoint you can have a secondary axis. 
- the line would have to have the data value written on top of it either in a different color or in black in smaller font so that it fits on top of the bar. 

Thank you so much for considering this. I think it would be helpful to other people that track measureable habits.
P. S. 
![Combination-chart-with-secondary-axis](https://github.com/user-attachments/assets/55b31201-35f5-44ba-a4bc-d0ce174daf12)

I found an image of a bar charts with a line chart on top as an example.

**Comment by MarkJeronimus** [comment]
Duplicate of https://github.com/iSoron/uhabits/discussions/1299
